app/controllers/posts_controller.rb

```diff
@@ -1,5 +1,37 @@
 class PostsController < ApplicationController
-  def index; end
+  def index
+    @user = User.find(params[:user_id])
+    @posts = @user.posts
+  end
 
-  def show; end
+  def show
+    @user = User.find(params[:user_id])
+    @post = @user.posts.find(params[:id])
+    @comments = @post.comments
+    @likes = @post.likes
+  end
+
+  def new
+    @current_user = current_user
+    @new_post = Post.new
+  end
```

Comment:
- [ ] Please create a method called `current_user` in your ApplicationController to make current user data available to all controllers. It should return the first user from the database as stated in the database. ℹ️ 

**- [ ] Please kindly ensure that comments and likes are successfully created for the current_user also.**